src/parsers/gas.ts

```diff
@@ -323,22 +323,27 @@ export const parseGasFees = (
   priceUnit: BigNumberish,
   nativeCurrency: string
 ) => {
-  const {
-    maxPriorityFeePerGas: maxPriorityFeePerGasGwei,
-    maxFeePerGas: maxFeePerGasGwei,
-  } = gasFeeParams || {};
-
-  const priorityFee = maxPriorityFeePerGasGwei?.amount || 0;
-  const maxFeePerGasAmount = maxFeePerGasGwei?.amount || 0;
+  const { maxPriorityFeePerGas, maxFeePerGas } = gasFeeParams || {};
+  const priorityFee = maxPriorityFeePerGas?.amount || 0;
+  const maxFeePerGasAmount = maxFeePerGas?.amount || 0;
   const baseFeePerGasAmount = baseFeePerGas?.amount || 0;
+
+  // if user sets the max base fee to lower than the current base fee
+  const estimatedFeePerGas = greaterThan(
+    maxFeePerGasAmount,
+    baseFeePerGasAmount
+  )
+    ? baseFeePerGasAmount
+    : maxFeePerGasAmount;
+
   const maxFee = getTxFee(
-    Number(maxFeePerGasAmount) + Number(priorityFee),
+    add(maxFeePerGasAmount, priorityFee),
```

Comment:
add accepts a bigNumberish and there's a scenario https://github.com/rainbow-me/rainbow/pull/2631/files#diff-d6624fa29be7d9dfa3819f22204ec01a0e468f7537771f4515740d6f3194be67R328 where `maxFeePerGasAmount = 0;` so you might wanna do a toString() on both parameters

Reply:
BigNumberish can be: number | string | BigNumber so we good for the 0